src/tsserver/webServer.ts

```diff
@@ -120,4 +145,28 @@ namespace ts.server {
         // Start listening
         session.listen();
     }
+
+    function startVirtualFileSystemSession(options: StartSessionOptions, logger: Logger, cancellationToken: ServerCancellationToken) {

```

Comment:
I think instead of this you would want to create session with optional option fsSystem which defaults to host. And then in Editor services and session you want to verify all FS events or on the FS host.. So we dont need two sessions and this can be done from any type of Session.

When testing you would want to send the VFS instance and test few operations that use FS.

Reply:
VFS instance should the host project and services use but server host is used for everything else.. So GC and other things work as before

Reply:
And then --vfs options can be parsed in both NodeServer as well as WebServer